content/client-lib-development-guide/features.textile

```diff
@@ -642,3 +655,13 @@ h4. CipherParams
 * @(TZ3)@ @Crypto#getDefaultParams@ function
 ** @(TZ3a)@ returns a complete @CipherParams@ instance with randomly generated key and iv
 ** @(TZ3b)@ takes an optional key parameter, returning a @CipherParams@ instance created with that key and with a keyLength calculated from that key
+
+h3(#defaults). Client Library defaults
+
+The following default values are configured for the client library:
+
+* @(DF1)@ Realtime defaults:
+** @(DF1a)@ @connectionStateTtl@ integer - default 60s. The duration that Ably will persist the connection state when a Realtime client is abruptly disconnected. When the client is in the @DISCONNECTED@ state, once this TTL has passed, the client should change the state to the @SUSPENDED@ state signifying that the state is now lost i.e. channels need to be reattached manually.
+Note that this default is override by any @connectionStateTtl@ specified in the @ConnectionDetails@ of the @CONNECTED@ @ProtocolMessage@
+** @(DF1b)@ @realtimeCloseTimeout@ - default 10s. Following an explicit request to close a connection from the client library by sending a @CLOSE@ @ProtocolMessage@, this value is the amount of time that the client library will wait before forcibly closing the connection and moving the client's connection state to @CLOSED@
```

Comment:
> Should we document it now, or leave that for 0.9 release?

No new timers in this spec version.

> we then probably need a different transport connect timeout which is different because the total time to connect and receive the first CONNECTED message is probably greater than the time expected for a MESSAGE, PRESENCE, ATTACH, DETACH, CLOSE message. WDYT?

I don't think it needs to be different. There's the round trip in each case, which is the same, and the realtime system response in each case should be "instantaneous" except for factors such as:

- establishing account and app master, which requires (potentially) cross-region round trip(s) and may also require a wait for the cluster to become stable if it is momentarily unstable. This applies to `CONNECT` and to all REST operations;
- establishing channel master (which suffers from the same issues and applies to `ATTACH` but not `CONNECT`);

- general delays due to processing loading (which affects both, but arguably channel operations more than `CONNECT` since it is affected by core load as well as frontend load.